### PR TITLE
Add counting sort

### PR DESCRIPTION
**Pull request by malayaleecoder**
@aktech : Have a look and suggest me further improvements.

### CODE REVIEW

**Comment by aktech** [comment]
@malayaleecoder  Thanks for the PR. Please resolve conflicts.

**Comment by malayaleecoder** [comment]
@aktech I think it would be fine now. I have already done gnome sort, will open a PR after you merge this :)

File: pydsa/__init__.py [hunk 1]
```diff
@@ -7,3 +7,4 @@
 from .sleep_sort import sleep_sort
 from .bfs import bfs
 from .binary_search import binary_search
+from .counting_sort import counting_sort
```
Comment (CuriousLearner) [inline pydsa/__init__.py]: Line ending missing

File: pydsa/counting_sort.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    count = [0] * (maxval + 1)
+    for a in array:
+    	count[a] += 1
+    i = 0
+    for a in range(maxval + 1):
+    	for c in range(count[a]):
+    		array[i] = a
+    		i +=1
+    return array
```
Comment (CuriousLearner) [inline pydsa/counting_sort.py]: Line ending missing here as well

File: pydsa/tests/test_counting_sort.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from pydsa.counting_sort import counting_sort
+from random import randint
+
+
+def test_counting_sort():
+	a = b = [randint(1, 100) for i in range(100)]
+	assert counting_sort(a, max(a)) == sorted(b)
```
Comment (CuriousLearner) [inline pydsa/tests/test_counting_sort.py]: and here as well

**Comment by malayaleecoder** [comment]
@CuriousLearner Done :)

**Comment by aktech** [comment]
Please add the reference used for implementing the algorithm at the top of the file.

**Comment by CuriousLearner** [comment]
@malayaleecoder Also, please fix the conflicts!

File: pydsa/tests/test_counting_sort.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from pydsa.counting_sort import counting_sort
+from random import randint
+
+
+def test_counting_sort():
+	a = b = [randint(1, 100) for i in range(100)]
```
Comment (aktech) [inline pydsa/tests/test_counting_sort.py]: pep8: W191 indentation contains tabs

File: pydsa/tests/test_counting_sort.py (hunk 3 above)
Comment (aktech) [inline pydsa/tests/test_counting_sort.py]: pep8: W191 indentation contains tabs

File: pydsa/counting_sort.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Reference : https://en.wikibooks.org/wiki/Algorithm_Implementation/Sorting/Counting_sort
+# Counting Sort
+# Complexity : O(n + (range of input))
+
+def counting_sort(array, maxval):
```
Comment (aktech) [inline pydsa/counting_sort.py]: pep8: E302 expected 2 blank lines, found 1

File: pydsa/counting_sort.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    >>> from pydsa import counting_sort
+    >>> a = [5, 6, 1, 9, 3]
+    >>> counting_sort(a, 9)
+    [1, 3, 5, 6, 9]
+
+    """
+    count = [0] * (maxval + 1)
+    for a in array:
+    	count[a] += 1
```
Comment (aktech) [inline pydsa/counting_sort.py]: pep8: E101 indentation contains mixed spaces and tabs

File: pydsa/counting_sort.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    i = 0
+    for a in range(maxval + 1):
+    	for c in range(count[a]):
```
Comment (aktech) [inline pydsa/counting_sort.py]: pep8: E101 indentation contains mixed spaces and tabs

File: pydsa/counting_sort.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    		array[i] = a
```
Comment (aktech) [inline pydsa/counting_sort.py]: Same here

File: pydsa/counting_sort.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    		i += 1
```
Comment (aktech) [inline pydsa/counting_sort.py]: same here

**Comment by aktech** [comment]
Please fix pep8 Errors.
Tip: Use a pep8 plugin to check for pep8 errors for your code editor.

**Comment by malayaleecoder** [comment]
@aktech : Sure, will do it ASAP. Thanks

**Comment by aktech** [comment]
Looks good. +1

**Pull request merged**